### ISSUE TITLE
Debug output gives crazy result

### ISSUE THREAD

**Issue by thomasj02** [mamrhein/fpdec.rs#6]
The debug output implementation seems broken. Here's a minimal reproducible example

This small test program:
```
fn main() {
    let decimal = fpdec::Decimal::new_raw(-42472286074876, 18);
    println!("{}", decimal);
    println!("{:?}", decimal);
}
```

Gives the output:
```
-0.000042472286074876
Dec!(-1.999957527713925124)
```

fpdec 0.8.0
rustc 1.72.0-nightly (04075b320 2023-06-22)

**Comment by mamrhein** [comment]
I added a test case:
```
    #[test]
    fn test_issue_6() {
        let d = Decimal::new_raw(4247228607487600, 18);
        let s = format!(concat!("Dec!", "({})"), d);
        let t = format!("{d:?}");
        assert_eq!(t, s);
    }
```
It doesn't fail.

Please try to force a rebuild by running `cargo clean`.

**Comment by thomasj02** [comment]
Can you check with rust nightly?

I created a completely clean project with cargo init and confirmed that the issue still exists

```
$ cargo init fpdec_bug
$ cargo add fpdec
    Updating crates.io index
      Adding fpdec v0.8.0 to dependencies.
             Features:
             + std
             - num-traits
             - packed
             - rkyv
             - serde-as-str
    Updating crates.io index
$ vi src/main.rs 
$ cargo run
  Downloaded proc-macro2 v1.0.63
  Downloaded 1 crate (44.9 KB) in 0.37s
   Compiling proc-macro2 v1.0.63
   Compiling unicode-ident v1.0.9
   Compiling quote v1.0.28
   Compiling fpdec-core v0.8.0
   Compiling fpdec-macros v0.8.0
   Compiling fpdec v0.8.0
   Compiling fpdec_bug v0.1.0 (/home/tjohnson/fpdec_bug)
    Finished dev [unoptimized + debuginfo] target(s) in 9.65s
     Running `target/debug/fpdec_bug`
-0.000042472286074876
Dec!(-1.999957527713925124)
$ cat src/main.rs 
fn main() {
    let decimal = fpdec::Decimal::new_raw(-42472286074876, 18);
    println!("{}", decimal);
    println!("{:?}", decimal);
}
$ rustc --version
rustc 1.72.0-nightly (04075b320 2023-06-22)
```

**Comment by mamrhein** [comment]
The defect is related to negative values.
I'll prep a fix soon.

**Comment by mamrhein** [comment]
Fixed in v0.8.1.
